modules/data-streams/src/yamlRestTest/resources/rest-api-spec/test/data_stream/220_failure_store_cluster_setting.yml

```diff
@@ -0,0 +1,243 @@
+          path: /_index_template/{template}
+          capabilities: [ 'failure_store_in_template' ]
+
+  - do:
+      cluster.put_settings:
+        body:
+          persistent:
+            data_streams.failure_store.enabled: '*-matching'
+
+  - do:
+      ingest.put_pipeline:
+        id: "failing_pipeline"
+        body:  >
+          {
+            "processors": [
+              {
+                "fail": {
+                  "message" : "pipeline go boom"
+                }
+              }
+            ]
+          }
+
+  - do:
+      indices.put_index_template:
+        name: index_template_default_fs
+        body:
+          index_patterns: default-fs-*
+          data_stream: {}
+          template:
+            settings:
+              number_of_shards:   1
+              number_of_replicas: 1
+
+  - do:
+      cluster.put_component_template:
+        name: component_template_disabled_fs
+        body:
+          template:
+            data_stream_options:
+              failure_store:
+                enabled: false
+
+  - do:
+      indices.put_index_template:
+        name: index_template_disabled_fs
+        body:
+          index_patterns: disabled-fs-*
+          data_stream: {}
+          composed_of:
+            - component_template_disabled_fs
+          template:
+            settings:
+              number_of_shards:   1
+              number_of_replicas: 1
+
+---
+teardown:
+  - do:
+      indices.delete_data_stream:
+        name: default-fs-matching
+        ignore: 404
+
+  - do:
+      indices.delete_data_stream:
+        name: disabled-fs-matching
+        ignore: 404
+
+  - do:
+      indices.delete_index_template:
+        name: index_template_disabled_fs
+        ignore: 404
+
+  - do:
+      cluster.delete_component_template:
+        name: component_template_disabled_fs
+        ignore: 404
+
+  - do:
+      indices.delete_index_template:
+        name: index_template_default_fs
+        ignore: 404
+
+  - do:
+      ingest.delete_pipeline:
+        id: "failing_pipeline"
+        ignore: 404
+
+  - do:
+      cluster.put_settings:
+        body:
+          persistent:
+            data_streams.failure_store.enabled: ''
+
+---
+"Redirect ingest failure when auto-creating data stream to failure store when enabled by setting":
+  - do:
+      index:
+        index:  default-fs-matching
+        refresh: true
+        pipeline: 'failing_pipeline'
+        body:
+          '@timestamp': '2020-12-12'
+          foo: bar
+  - match: { failure_store: used }
+  - match: { _index: '/\.fs-default-fs-matching-(\d{4}\.\d{2}\.\d{2}-)?\d{6}/' }
+
+---
+"Redirect ingest failure into pre-existing data stream to failure store when enabled by setting":
+  - do:
+      index:
+        index:  default-fs-matching
+        refresh: true
+        body:
+          '@timestamp': '2020-12-11'
+          foo: bar
+  - match: { failure_store: null }
+  - match: { _index: '/\.ds-default-fs-matching-(\d{4}\.\d{2}\.\d{2}-)?\d{6}/' }
```

Comment:
Nit: Maybe here we can just create the data stream with `indices.create_data_stream`.

Reply:
Definitely simplifies things and makes it more clear what we're actually testing. Thanks. (I'd looked to see if this action existed, but I think I'd been looking for something like `put_data_stream` rather than `create_` so I didn't find it, and figured I'd do it the long way around... I should have looked harder!)